t-tap/test_commands.c

```diff
@@ -509,6 +510,30 @@ void test_host_commands(void)
 	ok(!target_host->problem_has_been_acknowledged, "REMOVE_HOST_ACKNOWLEDGEMENT removes a host acknowledgement");
 	target_host->current_state = STATE_UP;
 
+	target_host->current_state = STATE_DOWN;
+	ok(CMD_ERROR_OK == process_external_command1("[1234567890] ACKNOWLEDGE_HOST_PROBLEM_EXPIRE;host1;2;0;0;4102441200;myself;expire in the future"), "core command: ACKNOWLEDGE_HOST_PROBLEM_EXPIRE");
+	ok(target_host->problem_has_been_acknowledged, "ACKNOWLEDGE_HOST_PROBLEM_EXPIRE with future end_time acknowledges a host problem");
+
+	ok(CMD_ERROR_OK == process_external_command1("[1234567890] REMOVE_HOST_ACKNOWLEDGEMENT;host1"), "core command: REMOVE_HOST_ACKNOWLEDGEMENT");
+	ok(!target_host->problem_has_been_acknowledged, "REMOVE_HOST_ACKNOWLEDGEMENT removes a host acknowledgement with expiry");
+	target_host->current_state = STATE_UP;
+
+	target_host->current_state = STATE_DOWN;
+	ok(CMD_ERROR_OK == process_external_command1("[1234567890] ACKNOWLEDGE_HOST_PROBLEM_EXPIRE;host1;2;0;0;946681200;myself;expire in the past"), "core command: ACKNOWLEDGE_HOST_PROBLEM_EXPIRE");
+	ok(!target_host->problem_has_been_acknowledged, "ACKNOWLEDGE_HOST_PROBLEM_EXPIRE with past end_time doesn't acknowledge a host problem");
+	target_host->current_state = STATE_UP;
+
+	clear_event_queue();
+	target_host->current_state = STATE_DOWN;
+	time(&current_time);
+	nm_asprintf(&cmdstr, "[1234567890] ACKNOWLEDGE_HOST_PROBLEM_EXPIRE;host1;2;0;0;%llu;myself;expire in two seconds", (unsigned long long int)current_time+2);
```

Comment:
Perhaps `1234567890` -> `current_time` now that we anyway have it.

Reply:
This isn't done in any of the other tests, but I did update the commit and removed the current_time variable; it's not used in other tests either.